src/sas/system/config/config_meta.py

```diff
@@ -74,20 +83,17 @@ def update(self, data: Dict[str, Any]):
                     logger.error(f"Cannot set set variable '{key}', improper type ({e.message})")
 
             else:
-                logger.error(f"Unknown config key: '{key}', skipping")
+                logger.warning(f"Unknown config key: '{key}', skipping")
+                self._bad_entries[key] = data[key]
 
     def save(self):
-        if self._write_disabled:
-            logger.error("Write disabled, this is probably because it will overwrite an outdated config.")
-            return
-
-        with open("config.json", 'w') as file:
+        with open(self.config_filename(True), 'w') as file:
             self.save_to_file_object(file)
 
     def save_to_file_object(self, file):
         """ Save config file
 
-        Only changed variables will be included in the saved file
+        Only changed and unknown variables will be included in the saved file
```

Comment:
Yes. That will happen. I think its worth it to not have "sticky" config entries. I know its not how you would like to see it done, and maybe it will turn out to be a bad way, but I've seen enough problems doing it the way you suggest to think its worth trying to do it differently.